src/flows/context/flow.list.tsx

```diff
@@ -296,7 +313,14 @@ export const FlowListProvider: FC = ({children}) => {
     const data = await getAllAPI(org.id)
     if (data && data.flows) {
       const _flows = {}
-      data.flows.forEach(f => (_flows[f.id] = hydrate(f.spec)))
+      data.flows.forEach(f => {
+        _flows[f.id] = {
+          ...hydrate(f.spec),
```

Comment:
not sure whether we should update the spec to include the createdAt and updatedAt metadata, or if that's a different issue

Reply:
i think that's a different issue, but you should def normalize how + when + where these values are being loaded into the data... maybe open up the hydrate function to handle the response from the api?

Reply:
Totally, I think I did open the hydrate function to do that, but in this case the spec doesn't contain an updatedAt and createdAt property. Should I just pass those properties in from the flow as something like:

```
hydrate({...f.spec, updatedAt: f.updatedAt, createdAt: f.createdAt})
```
or would the way to go be about updating the `spec`? I'm still fuzzy on where the source of truth for the spec is and if it's different to align with API expectations

Reply:
no.. the api should have a concept of a "notebook object" type right? of which, this is taking a properties of.. in this case `spec`.. i'm saying, don't update `spec`, but update the hydrate function to consume that higher order "notebook object" whatever that is

Reply:
Aw, gotcha. That makes sense...I think. I'll take a look at the code

Reply:
Honestly after trying to rework this, I'm not sure if that's the solution. It just adds unnecessary complexity to the hydrate function. I think if we had a definitively typed object that was being passed in but it seems like we're using the hydrate function in a couple places and it might be outta scope to type it here